### ISSUE TITLE
containerd installed with unsupported version when deploying on Hetzner Cloud

### ISSUE THREAD

**Issue by 7oku** [kubermatic/kubeone#2215]
### What happened?
Deploying a cluster (on Hetzner Cloud and Ubuntu 20.04) with kubernetes 1.23.7 results in containerd version 1.4.13 to be installed by kubeone instead of the available v1.6.6. 

```
NAME                                       STATUS   ROLES                  AGE    VERSION   INTERNAL-IP   EXTERNAL-IP       OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
pretty-k1-cluster-control-plane-1          Ready    control-plane,master   58m    v1.23.7   10.0.0.6      167.xxx.xxx.xxx   Ubuntu 20.04.4 LTS   5.4.0-122-generic   containerd://1.4.13
pretty-k1-cluster-control-plane-2          Ready    control-plane,master   58m    v1.23.7   10.0.0.5      168.xxx.xxx.xxx   Ubuntu 20.04.4 LTS   5.4.0-122-generic   containerd://1.4.13
pretty-k1-cluster-control-plane-3          Ready    control-plane,master   58m    v1.23.7   10.0.0.4      167.xxx.xxx.xxx   Ubuntu 20.04.4 LTS   5.4.0-122-generic   containerd://1.4.13
pretty-k1-cluster-pool1-56bf86bd86-cjzml   Ready    <none>                 56m    v1.23.7   10.0.0.9      167.xxx.xx.xxx    Ubuntu 20.04.4 LTS   5.4.0-122-generic   containerd://1.4.13
pretty-k1-cluster-pool1-56bf86bd86-nzmcx   Ready    <none>                 56m    v1.23.7   10.0.0.8      167.xxx.xx.xxx    Ubuntu 20.04.4 LTS   5.4.0-122-generic   containerd://1.4.13
```

There is a routine in the code to add the official docker repo, install the newest version of containerd and pin the release: https://github.com/kubermatic/kubeone/blob/06b17b7f7ab7baaef41ee60773ee0ebbed9750ca/pkg/scripts/render.go#L153 but this somehow is not succeeding, as the output clearly shows version 1.4 is being applied:

```
[168.xxx.xxx.xxx] Reading package lists...
[168.xxx.xxx.xxx] + sudo apt-mark unhold containerd.io
[168.xxx.xxx.xxx] containerd.io was already not hold.
[168.xxx.xxx.xxx] + sudo DEBIAN_FRONTEND=noninteractive apt-get install --option Dpkg::Options::=--force-confold --no-install-recommends -y 'containerd.io=1.4.*'
[168.xxx.xxx.xxx] Reading package lists...
[167.yyy.yyy.yyy] Reading package lists...
[167.yyy.yyy.yyy] + sudo apt-mark unhold containerd.io
[168.xxx.xxx.xxx] Building dependency tree...
[168.xxx.xxx.xxx] Reading state information...
[167.yyy.yyy.yyy] containerd.io was already not hold.
[167.yyy.yyy.yyy] + sudo DEBIAN_FRONTEND=noninteractive apt-get install --option Dpkg::Options::=--force-confold --no-install-recommends -y 'containerd.io=1.4.*'
[167.yyy.yyy.yyy] Reading package lists...
[168.xxx.xxx.xxx] The following NEW packages will be installed:
[168.xxx.xxx.xxx]   containerd.io
[168.xxx.xxx.xxx] 0 upgraded, 1 newly installed, 0 to remove and 2 not upgraded.
[168.xxx.xxx.xxx] Need to get 23.7 MB of archives.
[168.xxx.xxx.xxx] After this operation, 110 MB of additional disk space will be used.
[168.xxx.xxx.xxx] Get:1 https://download.docker.com/linux/ubuntu focal/stable amd64 containerd.io amd64 1.4.13-1 [23.7 MB]
[167.yyy.yyy.yyy] Building dependency tree...
[167.yyy.yyy.yyy] Reading state information...
[167.yyy.yyy.yyy] The following NEW packages will be installed:
[167.yyy.yyy.yyy]   containerd.io
[167.yyy.yyy.yyy] 0 upgraded, 1 newly installed, 0 to remove and 2 not upgraded.
[167.yyy.yyy.yyy] Need to get 23.7 MB of archives.
[167.yyy.yyy.yyy] After this operation, 110 MB of additional disk space will be used.
[167.yyy.yyy.yyy] Get:1 https://download.docker.com/linux/ubuntu focal/stable amd64 containerd.io amd64 1.4.13-1 [23.7 MB]
[168.xxx.xxx.xxx] Fetched 23.7 MB in 0s (54.9 MB/s)
[168.xxx.xxx.xxx] Selecting previously unselected package containerd.io.
[168.xxx.xxx.xxx] (Reading database ... 
```

We can circumvent this for the **worker** nodes created by the machine controller with the terraform option

```
operatingSystemSpec = {
          distUpgradeOnBoot = true
}
```
however the **control plane** nodes still will be deployed with the old version. 

v1.4.13 is EoL and not supported by current kubernetes versions any more. 

### Expected behavior

Either the latest containerd version should be installed or the version should be configurable in the kubeone.yml explicitely
for both, control plane and worker nodes.

### How to reproduce the issue?

Deploy cluster at Hetzner Cloud with this kubeone.yml given below. `kubeone apply -f kubeone.yml -t ../terraform/tf.json -v`

### What KubeOne version are you using?

```bash
# kubeone version
{
  "kubeone": {
    "major": "1",
    "minor": "4",
    "gitVersion": "1.4.5",
    "gitCommit": "a56d5566abef82bbc632b0356bcc9b72cfe752cf",
    "gitTreeState": "",
    "buildDate": "2022-07-12T09:29:32Z",
    "goVersion": "go1.18.1",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "43",
    "gitVersion": "v1.43.3",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

### Provide your KubeOneCluster manifest here (if applicable)

```yaml
apiVersion: kubeone.k8c.io/v1beta2
kind: KubeOneCluster
versions:
  kubernetes: '1.23.7'
cloudProvider:
  hetzner: {}
  external: true
clusterNetwork:
  kubeProxy:
    skipInstallation: true
  cni:
    cilium:
      enableHubble: true
      kubeProxyReplacement: strict
```

### What cloud provider are you running on?

Hetzner Cloud

### What operating system are you running in your cluster?

Ubuntu 20.04 image from Hetzner

**Comment by xmudrii** [comment]
This is as expected. KubeOne 1.4 uses containerd 1.4. The upcoming KubeOne 1.5 will use containerd 1.5. KubeOne currently doesn't support containerd 1.6 because there are some breaking API changes (see #2076 for more details).

I'll check tomorrow internally if we can upgrade containerd to 1.5 for KubeOne 1.4. containerd 1.6 will be probably supported only in KubeOne 1.6 (planned for the end of 2022 or the start of 2023).

**Comment by 7oku** [comment]
Got you! Seems i got confused by a the 3 months old bump in #2020 but just realized it is not incorporated in the quite new 1.4.5 as it seems to be based of 1.4 release branch. 

Bumping to containerd 1.6 is not required necessarily, kubernetes 1.23.x just needs a supported containerd version (1.5.8+ seems to be good for now), so we are looking forward either for KubeOne 1.5 or containerd v1.5 with KubeOne 1.4.x in the meantime.

**Comment by xmudrii** [comment]
This is fixed with the KubeOne 1.4.6 release: https://github.com/kubermatic/kubeone/releases/tag/v1.4.6
